membersonlyevent.php

```diff
@@ -223,3 +223,19 @@ function membersonlyevent_civicrm_copy($objectName, &$object) {
     $currentListener->handle($object);
   }
 }
+
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Implements hook_civicrm_container().
+ */
+function membersonlyevent_civicrm_container($container) {
+  $container->register(CRM_MembersOnlyEvent_Service_MembersOnlyEventAccess::class, CRM_MembersOnlyEvent_Service_MembersOnlyEventAccess::class);
+
+  $container->register(CRM_MembersOnlyEvent_Hook_PreProcess_Register::class, CRM_MembersOnlyEvent_Hook_PreProcess_Register::class)
```

Comment:
What would happen if you don't register `CRM_MembersOnlyEvent_Hook_PreProcess_Register`?
No, we need it this way so Symfony container will create only one instance of `MembersOnlyEventAccessService` which is used by two hooks in the same request so that I can execute any logic once.
Although initially, my goal was to improve the performance a little, I am not sure how things will  go in the production, and this approach is not quite popular in the CiviCRM community 
At the end I agree with you we need to think of a different.

Reply:
> and this approach is not quite popular in the CiviCRM community

I think it is not about being popular or not, it is more of it make sense or not, my problem with the approach above is that you are using service container not the way it is intended to be used (and I am not saying that I am an expert in it, it is just my thoughts given my current understanding), both hook implementation classes are not services, meaning they do not contain business logic and that they are going to be used in multiple places in the system, so defining them as services just to be able to pass information between classes seems like more of a hack to me and that's why I generally don't like the approach.

Reply:
You have a point.